Documentation/proposals/20250122-blackbox-crd.md

```diff
@@ -0,0 +1,52 @@
+## Your Proposal Title
+
+* **Owners:**
+  * [Pger-Y](https://github.com/Pger-Y)
+
+* **Related Tickets:**
+  * [Discussion](https://github.com/prometheus-operator/prometheus-operator/discussions/7134#discussioncomment-11569949)
+
+* **Other docs:**
+  * n/a
+
+This document proposes the creation of a Blackbox Operator and associated CRDs, similar to Prometheus-Operator, to manage blackbox-exporter instances and introduce a Module CRD for handling dynamic configuration of blackbox modules in a Kubernetes-native way.
```

Comment:
Hi @Pger-Y question, when you have "Kubernetes-native way", do you mean to be able to pick targets with selectors, or is it another thing I'm missing? Just because I've worked with the probe in the past and the downside at the time was the [static](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api-reference/api.md#monitoring.coreos.com/v1.ProbeTargetStaticConfig) config for targets.

Reply:
Ok thanks I was just considering the targets since for my specific use case there were targets inside and outside K8s, and maybe expanding on the current Probe CRD would be a good idea. But I'm guessing that should be better suited for an issue instead.